plasmapy/atomic/ionization_state.py

```diff
@@ -233,6 +235,17 @@ def __getitem__(self, value) -> State:
                     raise ChargeError("No integer charge provided.")
         return result
 
+    def __setitem__(self, key: Any, value: Any):
+        """
+        Raise a `NotImplementedError` because the ionic fractions for
+        different ionization levels should be set simultaneously due to
+        the normalization constraint.
+        """
+        raise NotImplementedError(
+            "Dictionary assignment of an IonizationState instance is "
+            "not allowed. because the ionic fractions of for different "
+            "ionization levels should be set simultaneously.")
```

Comment:
Looks like a good future issue!

Reply:
For this, I'm thinking that dictionary assignment should not be allowed because of the normalization constraint that the sum of ionic fractions must equal one.  If we changed the ionic fraction of 'Fe 2+', then we would have to change other levels too at the same time.  It's probably better to do something like:
```Python
>>> helium_states = IonizationState('He', [0.5, 0.3, 0.2])
>>> helium_states[2] = 0.4  # raises exception because of normalization constraint
>>> helium_states.ionic_fractions = [0.6, 0.3, 0.1]  # this should work fine
```